server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -268,7 +268,10 @@ public ThreadPool(
             new ScalingExecutorBuilder(Names.REMOTE_REFRESH, 1, halfProcMaxAt10, TimeValue.timeValueMinutes(5))
         );
         if (FeatureFlags.isEnabled(FeatureFlags.CONCURRENT_SEGMENT_SEARCH)) {
-            builders.put(Names.INDEX_SEARCHER, new FixedExecutorBuilder(settings, Names.INDEX_SEARCHER, allocatedProcessors, 1000, false));
+            builders.put(
```

Comment:
@jed326 I have a concern with this approach. The INDEX_SEARCHER pool right now is shared between all indices on the node (to be more precise, between all shards on the node, but that's a detail). It leads to unfairness: the search queries run over "hot" index (or indices) may fill up the pool and basically harm other searches. 

To me, the approach we may consider is looking slightly more complicated: consider per-index thread pools instead of a shared one. Or, alternatively:
 - consider the specialized thread pool implementation that  tries it best to schedule per-segment searches as fair as possible (balance between shards)
 - consider the specialized thread pool implementation that  tries it best to schedule per- segment searches as efficient as possible (prioritize ongoing indices vs new ones)

Reply:
> I agree with your concern but I believe this is beyond the scope of what I'm trying to accomplish here, as the unfairness issue is going to exist in both the fixed threadpool today and the resizable threadpool I am changing it to.

Thank you, fair enough, created https://github.com/opensearch-project/OpenSearch/issues/7583. I think using resizable queue should not have negative consequences comparing to fixed one.